src/components/DataList/DataList.jsx

```diff
@@ -1,19 +1,32 @@
 import PropTypes from "prop-types";
 
-export const DataList = ({ list }) => {
+export const DataList = ({ list, register, required }) => {
   return (
-    <div>
-      <input list="list-items" type="text" placeholder="Szukaj..." />
+    <>
+      <input
+        {...register(list.title, { required })}
```

Comment:
After I saw this `register` I wrote the comment before. You should understand now why.